docs: update docs prior 1.11.0 release

alvarobartt opened the pull request:
# Description

This PR contains the following linked PRs to update the documentation before releasing Argilla 1.11.0:
* #3236 
* #3237 

**Type of change**

- [X] Documentation update

alvarobartt committed:
docs: add `User` and `Workspace` management layers (#3236)

# Description

This PR updates the existing documentation for "User Management" to
avoid showcasing the user-management via the Argilla API, to show how to
use the `User` class in the Python client instead; as well as adding a
new "Workspace Management" section, and both `User` and `Workspace`
classed to be included in the Python reference.

## What's missing?

- [ ] Update `user_management.md`
- [ ] Make sure that there are no references to the `/api/users`
explicitly in any documentation entry

**Type of change**

- [X] Documentation update

---------

Co-authored-by: frascuchon <francis@argilla.io>

alvarobartt committed:
fix: `python_client.rst` ref for `FeedbackDataset`

alvarobartt committed:
fix: remove ref to `argilla.server` in `workspaces` SDK

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/workspace_management.md:
````diff
@@ -0,0 +1,127 @@
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
````
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/workspace_management.md:
````diff
@@ -0,0 +1,127 @@
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
````
```suggestion
rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```

alvarobartt replied:
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/workspace_management.md:
````diff
@@ -0,0 +1,127 @@
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
````
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/workspace_management.md:
````diff
@@ -0,0 +1,127 @@
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
````
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/workspace_management.md:
````diff
@@ -0,0 +1,127 @@
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
````
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/user_management.md:
````diff
@@ -181,50 +195,79 @@ Options:
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
````
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/user_management.md:
````diff
@@ -181,50 +195,79 @@ Options:
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
````
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/user_management.md:
```diff
@@ -242,115 +285,65 @@ User successfully created:
-}
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/user_management.md:
```diff
@@ -396,99 +389,24 @@ Migrating User with username 'daisy'
-]
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<OWNER_API_KEY>")
```
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

davidberenstein1957 commented on docs/_source/getting_started/installation/configurations/user_management.md:
````diff
@@ -150,7 +117,54 @@ Path: /path/to/alembic/versions/1769ee58fbb4_create_workspaces_users_table.py
+```python
+import argilla as rg
+
+rg.init(api_url="<API_URL>", api_key="<API_KEY>")
````
```suggestion
rg.init(api_url="<ARGILLA_API_URL>", api_key="<ARGILLA_API_KEY>")
```

alvarobartt committed:
Apply suggestions from code review

Co-authored-by: David Berenstein <david.m.berenstein@gmail.com>

alvarobartt committed:
Apply suggestions from code review

Co-authored-by: David Berenstein <david.m.berenstein@gmail.com>

davidberenstein1957 committed:
fix: resolved wrong automodule import for the FeedbackDataset

nataliaElv committed:
docs: update assignments code in tutorials and llm guides (#3237)

<!-- Thanks for your contribution! As part of our Community Growers
initiative 🌱, we're donating Justdiggit bunds in your name to reforest
sub-Saharan Africa. To claim your Community Growers certificate, please
contact David Berenstein in our Slack community or fill in this form
https://tally.so/r/n9XrxK once your PR has been merged. -->

# Description

Changes code snippets in the LLM practical guides and examples to apply
the changes in user and workspace management from version 1.11.0

**Type of change**

- [x] Documentation update


**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [x] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)
